internal/pkg/service/common/etcdop/op/atomic.go

```diff
@@ -82,11 +84,25 @@ func (v *AtomicOp[R]) WritePhaseOps() (out []HighLevelFactory) {
 		// There is no processor callback, we can pass write phase as is
 		out = append(out, v.writePhase...)
 	} else {
-		out = append(out, func(ctx context.Context) (Op, error) { return v.createWriteTxn(ctx) })
+		out = append(out, func(ctx context.Context) (Op, error) { return v.writeTxn(ctx) })
 	}
 	return out
 }
 
+// RequireLock to run the operation. Internally, an IF condition is generated for each registered lock.
+//
+// The lock must be locked during the entire operation, otherwise the NotLockedError occurs.
+// This signals an error in the application logic.
+//
+// If the local state of the lock does not match the state in the database (edge case), then the LockedError occurs.
+// There are no automatic retries. Depending on the kind of the operation, you may retry or ignore the error.
+//
+// The method ensures that only the owner of the lock performs the database operation.
+func (v *AtomicOp[R]) RequireLock(lock *concurrency.Mutex) *AtomicOp[R] {
+	v.locks = append(v.locks, lock)
+	return v
+}
+
```

Comment:
`*concurrency.Mutex` has method `IsOwner() etcd.Cmp`, so the lock can be checked in a transaction IF condition.

Each registered lock is checked in the READ and in the WRITE transaction.

See test for examples.

-------------------

`*concurrency.Mutex.IsOwner`:
https://github.com/etcd-io/etcd/blob/42f0cb9762cafa440a3f77884b0deb454ccb22c5/client/v3/concurrency/mutex.go#L150C17-L152

------------------------

https://etcd.io/docs/v3.5/dev-guide/api_concurrency_reference_v3/
> Lock acquires a distributed shared lock on a given named lock. On success, it will return a unique key that exists so long as the lock is held by the caller. **This key can be used in conjunction with transactions to safely ensure updates to etcd only occur while holding lock ownership.** The lock is held until Unlock is called on the key or the lease associate with the owner expires.